src/VecSim/spaces/CMakeLists.txt

```diff
@@ -1,8 +1,8 @@
 project(VectorSimilarity_Spaces)
 
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 include(FetchContent)
 option(BUILD_TESTING "" OFF)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 FetchContent_Declare(
 	cpu_features
 	GIT_REPOSITORY  https://github.com/google/cpu_features.git
```

Comment:
better to extract this into something like `build/cmake/cpu-features.cmake` and include it from here.